vrecord

```diff
@@ -925,6 +935,15 @@ if [ -z "${duration}" ] ; then
     read duration
 fi
 
+# confirm legal value for diration
+if [ -n "${duration}" ]; then
+real_regexp="[+-]?([0-9]+|[0-9]+\.[0-9]*|\.[0-9]+)"
+if [[ $duration =~ ^$real_regexp$ ]]; then
+is_number=yes
+else echo "Illegal value for recording time. Input must only be numbers." && exit
+fi
+fi
+
```

Comment:
The repeat of the function accommodates the different ways a user could enter the duration value either through the GUI or the CLI prompt.  If they leave the option blank in the GUI then vrecord will prompt the user for a value via CLI (and verify its validity) after they have input the identifier for the file to be created.

Having the function also after the GUI outputs means that if an illegal value is entered via the GUI vrecord will immediately warn the user rather than proceeding to ask for a file identifier and then aborting.